### PR TITLE
Add countermeasures JSON to API

### PR DESCRIPTION
**Pull request by wolverdude**
This PR adds a list of countermeasures to each applicable region in `data-CHANNEL-v4.json`. The added JSON looks like this:

```json
"Countermeasures": [
  {
    "tag": "sports cancellation",
    "description": "On 18 March, the Ministry of Interior Affairs banned...",
    "source": "https://en.wikipedia.org/wiki/2020_coronavirus_pandemic_in_Afghanistan",
    "start_date": "2020-03-18",
    "end_date": null,
    "quantity": 1.0
  },
  
  {
    "tag": "weddings canceled",
    "description": "On 18 March, the Ministry of Interior Affairs banned...",
    "source": "https://en.wikipedia.org/wiki/2020_coronavirus_pandemic_in_Afghanistan",
    "start_date": "2020-03-18",
    "end_date": null,
    "quantity": 1.0
  }
]
```

This depends on [Data PR #9](https://github.com/epidemics/epimodel-covid-data/pull/9)

### CODE REVIEW

File: epimodel/exports/epidemics_org.py [hunk 1]
```diff
@@ -532,6 +555,7 @@ def process_export(args) -> None:
             get_df_list(timezone_df, code),
             get_df_else_none(un_age_dist_df, m49),
             get_df_else_none(traces_v3_df, iso3),
+            get_df_else_none(countermeasures_df, code),
```
Comment (mathijshenquet) [inline epimodel/exports/epidemics_org.py]: I believe you can do `get_df_list` to ensure that you do not have to do L175-L177.  @hnykda knows better.

**Review by mathijshenquet — APPROVED** [review]
Otherwise looks good!

File: epimodel/exports/epidemics_org.py [hunk 2]
```diff
@@ -165,6 +169,13 @@ def extract_smallish_data(
         if traces_v3 is not None:
             d["TracesV3"] = traces_v3["TracesV3"]
 
+        if isinstance(countermeasures, pd.DataFrame):
+            d["Countermeasures"] = countermeasures.replace(
+                {np.nan: None}).to_dict(orient='records')
+        elif isinstance(countermeasures, pd.Series):
+            d["Countermeasures"] = [countermeasures.replace(
+                {np.nan: None}).to_dict()]
```
Comment (mathijshenquet) [inline epimodel/exports/epidemics_org.py]: See other comment

File: epimodel/exports/epidemics_org.py [hunk 3]
```diff
@@ -117,7 +120,7 @@ def __init__(
         self.current_estimate = current_estimate
         # Any per-region data. Large ones should go to data_ext.
         self.data = self.extract_smallish_data(
-            rates, hopkins, foretold, timezones, un_age_dist, traces_v3
+            rates, hopkins, foretold, timezones, un_age_dist, traces_v3, countermeasures
         )
         # Extended data to be written in a separate per-region file
         self.data_ext = self.extract_external_data(
```
Comment (mathijshenquet) [inline epimodel/exports/epidemics_org.py]: Could you add it to the external data instead?

**Review by mathijshenquet — CHANGES_REQUESTED** [review]

File: epimodel/exports/epidemics_org.py (hunk 3 above)
Comment (mathijshenquet) [reply epimodel/exports/epidemics_org.py]: I checked it and this doubles the main data size from 100kb -> 200kb so its actually fine. Lets do the splitting of in another PR where we also move the JH data to extfile.

**Review by mathijshenquet — APPROVED** [review]

File: epimodel/exports/epidemics_org.py [hunk 4]
```diff
@@ -478,6 +490,16 @@ def process_export(args) -> None:
...
+    countermeasures_df: pd.DataFrame = pd.read_csv(countermeasures)
+
+    def get_region_code_from_name(name):
+        regions = args.rds.find_all_by_name(name, ['country', 'subdivision'])
+        return regions[0].Code if regions else None
+
+    countermeasures_df['country'] = countermeasures_df.country.apply(
+        get_region_code_from_name)
+    countermeasures_df.set_index('country', inplace=True)
+
```
Comment (hnykda) [inline epimodel/exports/epidemics_org.py]: What about to add the `Code` into the countermeasures CSV directly so it's done only once and doesn't have to be with during the exporting + conforms to standards in other datasets? That would likely save you this whole code.

File: epimodel/exports/epidemics_org.py (hunk 1 above)
Comment (hnykda) [reply epimodel/exports/epidemics_org.py]: Yes, agree. By using get_df_list here would make sure you always return a dataframe and therfore you wouldn't have to do ugly: https://github.com/epidemics/epimodel/pull/45/files#diff-45d92a1ec7e6ad88e92722302782b9dbR175-R177

**Comment by hnykda** [comment]
Other than those comments, all good!

**Comment by mathijshenquet** [comment]
@wolverdude can we also expose the `countermeasure-tags.json` content to the frontend? see my PR #47

File: epimodel/exports/epidemics_org.py (hunk 1 above)
Comment (wolverdude) [reply epimodel/exports/epidemics_org.py]: I got an error when I tried to use `get_df_list`. But I'll look into it.

File: epimodel/exports/epidemics_org.py (hunk 4 above)
Comment (wolverdude) [reply epimodel/exports/epidemics_org.py]: The CSV is being created elsewhere, and I need the `Region` class to be able to get the code from the country, which is only available in this repo.

Comment (hnykda) [reply epimodel/exports/epidemics_org.py]: I know it is - but do we have the creation under control or not?

File: epimodel/exports/epidemics_org.py (hunk 1 above)
Comment (hnykda) [reply epimodel/exports/epidemics_org.py]: If you post which one, then we can help.

**Comment by wolverdude** [comment]
I will update this in the morning (later today for you European folks). We shouldn't merge until we're ready to have it on the frontend though, since it doubles the size of the main data file.

**Comment by wolverdude** [comment]
This is now ready for merging whenever it is needed.

**Comment by hnykda** [comment]
> I will update this in the morning (later today for you European folks). We shouldn't merge until we're ready to have it on the frontend though, since it doubles the size of the main data file.

Do we plan/have capacity to do the frontend? @mathijshenquet 

I would like not to wait too much so we don't get out of date branches

**Comment by mathijshenquet** [comment]
Yes, I plan on working on this tomorrow. Sorry for my late response. Thanks a lot for this @wolverdude!

**Comment by witzatom** [comment]
Whats the status of this PR? From what i understand:
* the notion countermeasure database, that this exposes is out of date, not sure whether there are any plans to work on it
* the paper recently published merges the notion database with a some other databases + human input
* the merged data might be a bit more up to date

**Pull request closed without merging**